p10k support, Affect dirstack and chwpd
--- michaelxmcbride---zsh-dircycle/dircycle.zsh	2020-10-14 03:42:40.565527743 -0400
+++ michaelxmcbride---zsh-dircycle/dircycle.zsh-patch	2020-10-14 03:42:32.315653673 -0400
@@ -5,20 +5,22 @@

     [[ ${#dirstack} -eq 0 ]] && return

-    while ! builtin pushd -q $1 &>/dev/null; do
+    while ! builtin pushd $1 &>/dev/null; do
         # A missing directory was found; pop it out of the directory stack.
-        builtin popd -q $1
+        builtin popd $1

         # Stop trying if there are no more directories in the directory stack.
         [[ ${#dirstack} -eq 0 ]] && break
     done
-
+
     # Trigger a prompt update if using Pure (https://github.com/sindresorhus/pure).
     if typeset -f prompt_pure_setup > /dev/null; then
         prompt_pure_async_tasks
         prompt_pure_preprompt_render
     else
-        zle reset-prompt
+        _p9k_precmd
+        zle .reset-prompt
+        zle -R
     fi
 }
